lf-edge/edge-home-orchestration-go
# [DataStorage] Revise configuration files in order to upload int, float, json data to EdgeX

- Use POST api like "IP:49986/api/v1/resource/datastorage/{int,float,json}" to upload data
- For example,
  - curl -X POST "localhost:49986/api/v1/resource/datastorage/int" -H "accept: text/plain" -H "Content-Type: text/plain" -d 123
  - curl -X POST "localhost:49986/api/v1/resource/datastorage/float" -H "accept: text/plain" -H "Content-Type: text/plain" -d 123.21
  - curl -X POST "localhost:49986/api/v1/resource/datastorage/json" -H "accept: application/json" -H "Content-Type: application/json" -d {"name":"hello"}

Signed-off-by: Taewan Kim <t25.kim@samsung.com>

# Description

Use POST api like "IP:49986/api/v1/resource/datastorage/{int,float,json}" to upload numeric and json data.

## Type of change


# How Has This Been Tested?
1. Run edgex containers by docker-compose like below.
```
$ docker-compose -f deployments/datastorage/docker-compose.yml up -d
```
2. Copy DataStorage configuration files(configuration.toml and datastorage-device.yaml) to /var/edge-orchestration/datastorage/ 
3. Run `edge-orchestration`.
4. Send some requests to upload data like below
```
$ curl -X POST "localhost:49986/api/v1/resource/datastorage/int" -H "accept: text/plain" -H "Content-Type: text/plain" -d 123
$ curl -X POST "localhost:49986/api/v1/resource/datastorage/float" -H "accept: text/plain" -H "Content-Type: text/plain" -d 123.21
$ curl -X POST "localhost:49986/api/v1/resource/datastorage/json" -H "accept: application/json" -H "Content-Type: application/json" -d {"name":"hello"}
```
5. Check the results from edgex-core-data via a web browser like below. 
    - url : http://localhost:48080/api/v1/event/device/datastorage/3 

## Result
![image](https://user-images.githubusercontent.com/5847128/117604054-68b40500-b18f-11eb-8f78-766194b1978c.png)


**Test Configuration**:
* Firmware version: (OS type and version, e.g., Ubuntu 14.04, etc.)
* Hardware: x86-64
* Edge Orchestration Release: Coconut

# Checklist:
